.github/workflows/release.yml

```diff
@@ -5,141 +5,246 @@ on:
+          IMAGE_NAME=darwinia
+          IMAGE_PREFIX=quay.io/darwinia-network
+          IMAGE_PUSH_NAME_TAG=${IMAGE_PREFIX}/${IMAGE_NAME}:${GIT_TAG_NAME}
+          IMAGE_PUSH_NAME_SHA=${IMAGE_PREFIX}/${IMAGE_NAME}:sha-${SHA}
 
+          docker build \
+            -t ${IMAGE_NAME} \
+            -f .maintain/docker/Dockerfile \
+            .
+
+          docker tag ${IMAGE_NAME} ${IMAGE_PUSH_NAME_TAG}
+          docker tag ${IMAGE_NAME} ${IMAGE_PUSH_NAME_SHA}
+
+          docker push ${IMAGE_PUSH_NAME_TAG}
+          docker push ${IMAGE_PUSH_NAME_SHA}
 
   complete-build:
     name: Complete build
     runs-on: ubuntu-latest
-    needs: [build-image, build-wasm]
+    needs: [build-subwasm, build-image]
     steps:
       - uses: actions/checkout@v2
 
       - uses: actions/download-artifact@v2
         with:
           name: darwinia-artifact
-          path: deploy
+          path: shared
+
+      - name: Prepare deploy folder
+        run: |
+          chmod +x shared/darwinia
 
-      - name: Complete build
-        run: .maintain/complete-build.sh
+          mkdir deploy
+          cd shared
+          tar cf darwinia.tar.zst darwinia -I pzstd
+          mv darwinia.tar.zst ../deploy/
 
-      - name: Publish github release
-        uses: docker://antonyurchenko/git-release:latest
-        if: startsWith(github.ref, 'refs/tags/v')
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          DRAFT_RELEASE: "false"
-          PRE_RELEASE: "false"
-          CHANGELOG_FILE: "CHANGELOG.md"
-          ALLOW_EMPTY_CHANGELOG: "false"
+          mv crab-runtime/*.compact.wasm ../deploy/
+          mv crab-runtime/*.compact.json ../deploy/
+          mv darwinia-runtime/*.compact.wasm ../deploy/
+          mv darwinia-runtime/*.compact.json ../deploy/
+
+## not open for now. https://github.com/darwinia-network/darwinia/pull/779#issuecomment-964911852
+#          mv crab-runtime/*.compact.compressed.wasm ../deploy/
+#          mv crab-runtime/*.compact.compressed.json ../deploy/
+#          mv darwinia-runtime/*.compact.compressed.wasm ../deploy/
+#          mv darwinia-runtime/*.compact.compressed.json ../deploy/
+
+      - name: Hash file
+        run: |
+          cd deploy
+          md5sum * > ../md5sums.txt
+          mv ../md5sums.txt ./
+
+      - uses: olegtarasov/get-tag@v2.1
+        id: tag-name
+      - uses: benjlevesque/short-sha@v1.2
+        id: short-sha
         with:
-          args: |
-            deploy/bin/*
+          length: 7
 
-      - uses: 8398a7/action-slack@v3
-        if: failure()
+      - name: Generate template data
+        run: |
+          PROPOSAL_CRAB_COMPACT=$(cat deploy/crab-runtime.compact.json | jq -r .proposal_hash)
+          PROPOSAL_DARWINIA_COMPACT=$(cat deploy/darwinia-runtime.compact.json | jq -r .proposal_hash)
+
+          # # not open for now. https://github.com/darwinia-network/darwinia/pull/779#issuecomment-964911852
```

Comment:
Two sharps.

Reply:
And the indents here don't consist with the other `not open for now.` comments. Please make sure to use a consistent indentation style.